engine/src/CalciteInterpreter.cpp

```diff
@@ -379,11 +379,11 @@ std::unique_ptr<ral::frame::BlazingTable> execute_plan(std::vector<ral::io::data
 		};
 		ral::batch::OutputKernel output;
 
-		auto graph = tree.build_batch_graph(logicalPlan);
-		if (graph.num_nodes() > 0) {
-			graph += link(graph.get_last_kernel(), output, ral::cache::cache_settings{.type = ral::cache::CacheType::CONCATENATING});
-			// graph.show();
-			graph.execute();
+		auto query_graph = tree.build_batch_graph(logicalPlan);
+		if (query_graph->num_nodes() > 0) {
+			*query_graph += link(query_graph->get_last_kernel(), output, ral::cache::cache_settings{.type = ral::cache::CacheType::CONCATENATING});
```

Comment:
i think this calls the copy constructor. is this signle access or is it possible another thread is accessing this at the same time? if so this can cause race conditions

Reply:
What is "this"? this calls the copy contructor? what line? what operation?  What is single access? The function execute_plan is only called once. The graph will be accessed later by multiple threads, but the graph once created here should not change

Reply:
line 384 sorry i must have missed clicked